Document correct nodes usage (#1062)
SAP/luigi: docs/navigation-configuration.md

...
```

### Best practices for navigation structuring

<!-- add-attribute:class:warning -->

> >**NOTE:** Do not use `getConfig` and `setConfig` methods to extend `navigation.nodes` configuration. Unwanted side effects might occur.

Reviewer: ```suggestion
>**NOTE:** Do not use the [Core API](https://docs.luigi-project.io/docs/luigi-core-api?section=luigi-config)`getConfig` and `setConfig` methods to extend `navigation.nodes` configuration. Unwanted side effects might occur.
```

> We encourage you to use (async) functions to build the node tree, since there are limitations if you try to update the `navigation.nodes` with `setConfig` and `configChanged('navigation')`. Each `node.children` which is of type function, gets executed every time it's parent node is routed to.

Reviewer: Can they only use async functions or both async and non-async? I'm not sure what you meant with the parenthesis. 

```suggestion
We encourage you to use functions or [async functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) to build the node tree, since there are limitations if you try to update the `navigation.nodes` with `setConfig` and `configChanged('navigation')`. Each `node.children` which is of type function gets executed every time its parent node is routed to.
```

> In the following example, the `settings.children` function will be excuted whenever the user navigates to `/settings` or one of its nested routes like `/settings/general`. In case you are fetching them from eg. a RESTful API, you need to take care of caching by yourself, if necessary.

Reviewer: ```suggestion
In the following example, the `settings.children` function will be executed whenever the user navigates to `/settings` or one of its nested routes such as `/settings/general`. In case you are fetching them from, for example, a RESTful API, you need to take care of caching by yourself, if necessary.
```

...
```

### Best practices for navigation structuring

<!-- add-attribute:class:warning -->
>**NOTE:** Do not use the [Core API](luigi-core-api.md#luigi-config) `Luigi.getConfig` and `setConfig` methods to extend `navigation.nodes` configuration. Unwanted side effects might occur.

We encourage you to use functions or [async functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) to build the node tree, since there are limitations if you try to update the `navigation.nodes` with `setConfig`. Each `node.children` which is of type function gets executed one time, once its parent node is routed to. The results are cached until we receive [`Luigi.configChanged('navigation.nodes')`](luigi-core-api.md#configChanged) event.

> In the following example, the `settings.children` function will be executed once the user navigates to `/settings` or one of its nested routes such as `/settings/general`. When navigating to other routes and then back to `/settings`, we are showing a chached variant until we explicitly execute `Luigi.configChanged('navigation.nodes')`.

Reviewer 2: Typo: chached -> cached
